lib/phoenix/live_dashboard/helpers/live_helpers.ex

```diff
@@ -19,18 +21,21 @@ defmodule Phoenix.LiveDashboard.LiveHelpers do
   def assign_defaults(socket, params, session, refresher? \\ false) do
     param_node = Map.fetch!(params, "node")
     found_node = Enum.find(nodes(), &(Atom.to_string(&1) == param_node))
+    target_node = found_node || node()
 
     socket =
       Phoenix.LiveView.assign(socket, :menu, %{
         refresher?: refresher?,
         action: socket.assigns.live_action,
-        node: found_node || node(),
+        node: target_node,
         metrics: session["metrics"],
-        os_mon: Application.get_application(:os_mon),
-        request_logger: session["request_logger"]
+        os_mon: TargetNode.os_mon(target_node),
+        request_logger: session["request_logger"],
+        dashboard_running?: TargetNode.dashboard_running?(target_node)
       })
 
     if found_node do
+      TargetNode.ensure_loaded(found_node, Phoenix.LiveDashboard.SystemInfo)
```

Comment:
You can merge those two operations into a single one that returns a boolean if the dashboard is running or not.

Reply:
Actually the three operations: os_mon, dashboard_running?, and the code loading. :)

Reply:
To be clear, the two operations are `dashboard_running?` and `ensure_loaded`. :) Perhaps a more structured approach is to have `ensure_loaded` return a `capabilities` field. The struct has three values:

```elixir
%{system_info: boolean(), os_mon: boolean(), dashboard: boolean()}
```

Reply:
Good idea, I guess more capabilities might be added in the future. What would the `system_info` be for though?

We could save on rpc's and include a `dashboard_capabilities_callback` in `SystemInfo`? Not sure it if belongs there though.

Reply:
You are right on both counts.

1. we don't need a system_info capabilities
2. we could have a function in system_info that returns the version and all capablities, reducing the number of calls

:+1: